src/main/kotlin/gg/skytils/skytilsmod/features/impl/misc/SlayerFeatures.kt

```diff
@@ -231,13 +245,14 @@ object SlayerFeatures : CoroutineScope {
         if (!Utils.inSkyblock) return
         if (event.phase != TickEvent.Phase.START || mc.theWorld == null || mc.thePlayer == null) return
         lastTickHasSlayerText = hasSlayerText
-        hasSlayerText = sidebarLines.any { it == "Slay the boss!" }
+        val index = sidebarLines.indexOf("Slay the boss!")
+        hasSlayerText = index != -1
         if (!lastTickHasSlayerText && hasSlayerText) {
-            val currentTier =
-                sidebarLines.find { it.startsWith("Voidgloom Seraph") }
-                    ?.substringAfter("Voidgloom Seraph")?.drop(1)
-                    ?: ""
-            expectedMaxHp = BossHealths["Voidgloom"]?.get(currentTier) ?: 0
+            sidebarLines.elementAtOrNull(index - 1)?.let {
+                val boss = it.substringBeforeLast(" ")
+                val tier = it.substringAfterLast(" ")
+                expectedMaxHp = BossHealths[boss.substringBefore(" ")]?.get(tier) ?: 0
```

Comment:
Why not put the `substringBefore` with the declaration?

Reply:
Forgot to change that after committing, will be fixed after I rework the relevantEntity and relevantColor getters or I do some profiler tests